modules/setup/library/Setup/WebWizard.php

```diff
@@ -259,6 +245,12 @@ public function setupPage(Form $page, Request $request)
                     unset($pageData['setup_config_db_resource']);
                     unset($pageData['setup_config_db_creation']);
                 }
+            } elseif ($authData != null) {
+                // If you validate the database and then come back to change the authentication type
+                // but do not change it, you will get an database configuration related error message on the next page.
+                // To avoid this error, the 'setup_config_db_resource' page must be unset.
```

Comment:
I can't reproduce this. Not on the master and not with your change here undone. Please make sure you are really sure that this is necessary.

Reply:
This is necessary if user choose external as Auth type,  validate database and then come back to change Auth type, but do not change.
I saw that user must validate DB again if he come back (if database was selected as Auth), so I set the if condition for all. We can change it to just external if needed